### PR TITLE
Fix Lexer 2 compatibility

### PR DESCRIPTION
**Pull request by derrabus**
This PR fixes an incompatibility with Lexer 2 due to an unnecessarily overridden property.

More details in the comments below.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -16,9 +16,14 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        php: [7.2, 7.3, 7.4, '8.0', 8.1, 8.2]
```
Comment (derrabus) [inline .github/workflows/tests.yml:19]: PHP 7.2-8.0 are not supported on this branch, so let's not test against them.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -16,9 +16,14 @@ jobs:
...
+        php: ['8.1', '8.2']
+        deps: [highest]
 
-    name: PHP ${{ matrix.php }}
+        include:
+          - php: '8.1'
+            deps: lowest
```
Comment (derrabus) [inline .github/workflows/tests.yml:24]: This new Job tests with low dependencies. This way, issues like the one I've just fixed can be spotted more easily.

File: src/EmailLexer.php [hunk 3]
```diff
@@ -144,13 +144,6 @@ class EmailLexer extends AbstractLexer
      */
     public Token $current;
 
-    /**
-     * The next token in the input.
-     *
-     * @var Token<int, string>|null
-     */
-    public Token|null $lookahead;
```
Comment (derrabus) [inline src/EmailLexer.php:152]: This property exists on the parent class and we don't need to override it. The problem is that in Lexer 2, there is not property type declared while on Lexer 3 there is.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (egulias) [reply .github/workflows/tests.yml:24]: If we add `lowest` to `matrix.deps` should that work for both versions?

Comment (derrabus) [reply .github/workflows/tests.yml:24]: Yes. Do you want a deps=lowest job for both PHP versions? I felt like one of those is enough.

**Comment by egulias** [comment]
Hi @derrabus!
Scrutinizer is no longer able to find code coverage. I believe that we need to change `.scrutinizer.yml` and add `file: build/logs/clover.xml` could fix it.

**Comment by derrabus** [comment]
Let's try it.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (egulias) [reply .github/workflows/tests.yml:24]: Ok, yes, one is enough.

**Comment by egulias** [comment]
Nope. I will work on it, I might need to do multiple test. Will merge and fix it later... in 3.x was working.

**Pull request merged**